subsys/jwt/jwt.c

```diff
@@ -154,23 +159,17 @@ static struct json_obj_descr jwt_payload_desc[] = {
  */
 static void jwt_add_header(struct jwt_builder *builder)
 {
-	static const struct jwt_header head = {
-		.typ = "JWT",
-#ifdef CONFIG_JWT_SIGN_RSA
-		.alg = "RS256",
-#endif
-#ifdef CONFIG_JWT_SIGN_ECDSA
-		.alg = "ES256",
-#endif
-	};
+	int jwt_header_len = ARRAY_SIZE(jwt_header);
 
-	int res = json_obj_encode(jwt_header_desc, ARRAY_SIZE(jwt_header_desc),
-				  &head, base64_append_bytes, builder);
-	if (res != 0) {
-		/* Log an error here. */
-		return;
+	strncpy(builder->buf, jwt_header, builder->len);
```

Comment:
I wonder if we shouldn't just move the string copy into the if statement blow (in which case, strcpy is fine). The JWT isn't going to work in the overflow case, so it doesn't help us to copy the buffer. And this will avoid potential future scans that may trigger strncpy as an unsafe function.